### DISCUSSION TITLE
Additional render when using Controller vs inputRef for material UI component

### DISCUSSION THREAD

**Discussion by dakshsaraf** [react-hook-form/react-hook-form#2485] [Q&A] [answered]
This link shows two inputs.
1) Email passes register via inputRef of Material UI
2) Username uses the Controller Component

Username renders an additional time compared to the Email component. Was just curious why does this happen.
Below is the codesandbox link

https://codesandbox.io/s/react-hook-form-renders-umzuv?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
They should be the same in terms of how we trigger the re-render.

**Reply by dakshsaraf** [reply]
Then why does controller re-render an additional time is it due to validation check of other fields?

**Reply by bluebill1049** [reply]
remove devtool.

**Reply by dakshsaraf** [reply]
that makes sense, thanks
